### ISSUE TITLE
race condition can cause missed events

### ISSUE THREAD

**Issue by codekitchen** [shokai/websocket-client-simple#12]
Because the Client currently spawns the socket reading thread in the object initializer, there's a small window where the user of this library hasn't had a chance to set up their event handlers yet, but the connection can emit events. For illustration:

```ruby
ws = WebSocket::Client::Simple.connect 'ws://example.com:8888'

# a :message or :open event could be emitted in this window of time, before we
# have a chance to register the handler below, and it will be lost permanently

ws.on :open do |msg|
  puts msg.data
end
```

Possible solutions include:

```ruby
# the connection thread is not actually started until this method is called, after we
# have a chance to register our handlers
ws.start
```

or:

```ruby
ws = WebSocket::Client::Simple.connect('ws://example.com:8888') do |ws|
  # we have a chance to register our handlers during the constructor,
  # before the thread is started
  ws.on :open do |msg|
    puts msg.data
  end
end
```

**Comment by shokai** [comment]
thank you for suggestion.
I have implemented block-style. Now `WebSocket::Client::Simple.connect` runs a given block before connecting WebSocket.

shipped with `v0.3.0`

**Comment by codekitchen** [comment]
Awesome, thanks
